sycl-jit/jit-compiler/lib/rtc/DeviceCompilation.cpp

```diff
@@ -197,12 +252,19 @@ jit_compiler::compileDeviceCode(InMemoryFile SourceFile,
   FixedCompilationDatabase DB{".", CommandLine};
   ClangTool Tool{DB, {SourceFile.Path}};
 
+  IntrusiveRefCntPtr<DiagnosticOptions> DiagOpts{new DiagnosticOptions};
+  ClangDiagnosticWrapper Wrapper(BuildLog, DiagOpts.get());
+  Tool.setDiagnosticConsumer(Wrapper.consumer());
+
   // Set up in-memory filesystem.
   Tool.mapVirtualFile(SourceFile.Path, SourceFile.Contents);
   for (const auto &IF : IncludeFiles) {
     Tool.mapVirtualFile(IF.Path, IF.Contents);
   }
 
+  // Suppress message "Error while processing" being printed to stdout.
+  Tool.setPrintErrorMessage(false);
+
```

Comment:
Super nit: This feels related to the diagnostics setup, so move before VFS setup.